SystemReference/Payload/Camera/Camera.cpp

```diff
@@ -6,117 +6,304 @@
 
 #include "Fw/Types/BasicTypes.hpp"
 #include <SystemReference/Payload/Camera/Camera.hpp>
+#include <libcamera/libcamera.h>
+#include <errno.h>
+#include <unistd.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <fstream>
 
 namespace Payload {
 
-// ----------------------------------------------------------------------
-// Construction, initialization, and destruction
-// ----------------------------------------------------------------------
+  // ----------------------------------------------------------------------
+  // Construction, initialization, and destruction
+  // ----------------------------------------------------------------------
 
-Camera ::Camera(const char *const compName)
-    : CameraComponentBase(compName), m_photoCount(0){}
+  Camera ::Camera(const char *const compName): CameraComponentBase(compName), m_photoCount(0){}
 
-void Camera ::init(const NATIVE_INT_TYPE queueDepth,
-                   const NATIVE_INT_TYPE instance) {
-  CameraComponentBase::init(queueDepth, instance);
-}
-
-bool Camera::open(I32 deviceIndex) {
-#ifdef USES_OPENCV
-  if (m_capture.isOpened()) {
-    this->log_ACTIVITY_LO_CameraAlreadyOpen();
-    return true;
+  void Camera ::init(const NATIVE_INT_TYPE queueDepth, const NATIVE_INT_TYPE instance) {
+    CameraComponentBase::init(queueDepth, instance);
+    camManager = std::make_unique<libcamera::CameraManager>();
```

Comment:
We should consider moving this statement out of `init`.  We are removing those across the board, and thus this would break that drive.  Given it takes no arguments, it is arguable that it is ok.

Reply:
Could it go in open?